File: modules/angular2/src/facade/workers.dart
library angular.core.facade.workers;

import "dart:html";
export 'dart:html' show Worker;

class WorkerWrapper{
  static Worker create(String url){
    return new Worker(url + ".js");
  }
  static void onMessage(Worker worker, fn(data)){
    worker.onMessage.listen((data){
      fn(data);
    });
  }
  static void postMessage(Worker worker, message){
    worker.postMessage(message);
  }
  static void terminate(Worker worker){
    worker.terminate();
  }
}

File: modules/angular2/src/facade/workers.ts
import {global} from 'angular2/src/facade/lang';
export declare var Worker;

export class WorkerWrapper{
  static create(url: string): Worker{
    return new Worker(url + ".js");
  }
  static onMessage(worker: Worker, fn: EventListener): void{
    worker.addEventListener("message", fn);
  }
  static postMessage(worker: Worker, message: Object): void{
    worker.postMessage(message);
  }
  static terminate(worker): void{
    worker.terminate();
  }
}
